### ISSUE TITLE
Machine Learning Model Deployment to AKS Fails when using Cluster Autoscaler

### ISSUE THREAD

**Issue by phehr2** [Azure/azure-sdk-for-python#29902]
- **Package Name**: azureml-core
- **Package Version**: v1
- **Operating System**: N/A
- **Python Version**:  3.7.7

**Describe the bug**
Unable to deploy models to AKS via the Python SDK `azureml.core.Model.deploy` when the [AKS cluster Autoscaler](https://learn.microsoft.com/en-us/azure/aks/cluster-autoscaler) is enabled. The deployment times out after 5 minutes before the autoscaler has a chance to scale out to support this new workload.

**To Reproduce**
1. Stand up an AKS cluster with autoscaler enabled. Set the minimum nodes to 1 and the max is at least 2
2. After the cluster starts, ensure that the number of running nodes is less than the max. If they are equal, update the autoscaler rules by incrementing the max nodes so that it's at least 1 higher than the current number of running nodes
3. Attach the new cluster as Kubernetes Compute in Azure ML
4. Create an Azure ML model
5. Deploy the new Azure ML model to the new cluster using `azureml.core.Model.deploy`. Ensure that the `AksWebservice.deploy_configuration` has `cpu_cores` and/or `memory_gb` values set high enough such that AKS would not be able to schedule the model onto the existing single node (due to the system resources running there already), but low enough that they can fit on whatever SKU size you selected in step 1. If staged properly, it should trigger the cluster autoscaler to begin adding a new  node
7. Watch as the AKS autoscaler kicks in and begins adding another node
8. Watch as the Azure ML deployment fails at the 5 minute mark, with a message: `Couldn't Schedule because the kubernetes cluster didn't have available resources after trying for 00:05:00`
9. Watch as the AKS autoscaler finishes adding the new node, just a little too late

**Expected behavior**
The deployment should wait for longer than 5 minutes before timing out, if the autoscaler is actively adding an additional node to support the requested workload.

**Screenshots**
N/A

**Additional context**
This [doc ](https://learn.microsoft.com/en-us/azure/machine-learning/v1/how-to-troubleshoot-deployment?tabs=azcli#container-cant-be-scheduled) tells me it's due to resource constraints, which is true. But the real problem is that deployment isn't waiting long enough for the AKS node pool to scale out; which (for us) takes just a bit longer than 5 minutes to scale out.

This could be worked around by making the timeout an optional parameter to the `deploy` function via the `deployment_config` parameter, with a default value of 5 minutes.

**Comment by xiangyan99** [comment]
@azureml-github @Azure/azure-ml-sdk

**Comment by hsonetta** [comment]
Hi, I am facing the same issue as exactly described above. What's the status on this issue?

**Comment by phehr2** [comment]
Hi @jaga-work, thank you for the response. I am reaching out to our BI team to see if we can upgrade to the latest SDK and set up an environment where we can reproduce. But unless the SDK has since been modified to allow the AKS cluster more than 5 minutes to scale out (or AKS has since been modified to scale out faster than 5 minutes), then I assume we will still be able to reproduce the issue.

**Comment by jaga-work** [comment]
@phehr2, we will be looking forward to your further updates.

**Comment by jaga-work** [comment]
If any user is facing this issue, kindly let us know the package version and other details.

**Comment by phehr2** [comment]
This issue was reported over a year and a half ago. We've since worked around it by making sure our clusters are large enough to accommodate the deployments without having to scale out in real-time.

Unfortunately, priorities and resources have changed within that length of time and we don't can't invest the time/effort right now to set up a sandbox environment to try to reproduce this.

But if (a) nothing was changed to address this issue since it was first reported, and (b) @hsonetta has reported this being an issue, then we should be able to assume that the issue is still happening. Has the AML team not been able to reproduce, or have they not tried to reproduce yet?
